workflow/executor/pns/pns.go

```diff
@@ -369,13 +370,19 @@ func (p *PNSExecutor) updateCtrIDMap() {
 	}
 }
 
+var backoffOver30s = wait.Backoff{
+	Duration: 1 * time.Second,
+	Steps:    7,
+	Factor:   2,
+}
+
 func (p *PNSExecutor) GetTerminatedContainerStatus(containerID string) (*corev1.Pod, *corev1.ContainerStatus, error) {
 	var pod *corev1.Pod
 	var containerStatus *corev1.ContainerStatus
-	err := wait.Poll(1*time.Second, 3*time.Second, func() (bool, error) {
+	err := wait.ExponentialBackoff(backoffOver30s, func() (bool, error) {
```

Comment:
under high load the Kubernetes API can cease to respond within 3s - this mitigates that